### PR TITLE
Added ability to choose configuration file name

### PR DESCRIPTION
**Pull request by mamatopian**
#### Description

I needed to use multiple configuration files for single Launcher (running multiple strategies in same time, multi-process) so I'm now starting Launcher in two ways possible:

`Launcher.exe <configFilename>`

or simply by

`Launcher.exe` when Launcher grabs default config file.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/ect)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [ ] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description` or `feature-<issue#>-<description>`

### CODE REVIEW

**Comment by jaredbroad** [comment]
Welcome @mamatopian! I have a long term vision of having this feel more of a CLI e.g.

`qc --mode=backtest --config="/configs/config.json" --algorithm="/algos/BasicTemplateAlgorithm.py" --parameters=emaSlow:200,emaFast:50 --output=results.json`

Would you be OK tweaking the launcher.exe edits to not rely on args[0] but "parse" the args array via a class? say `LeanArgumentParser` which should live in the Config project.

```
if (args.Length > 0) 
{
     Config.SetCliParameters( new LeanArgumentParser(args) );
}
```

Then inside the LeanArgumentParser we can create config overrides and all the system will continue to work from the Config class as it is now.

**Comment by jaredbroad** [comment]
Thanks :) Recommend not trying to implement all the switches, just your one: --config="file.json" then we can layer switches in over time to a solid foundation.

**Comment by mamatopian** [comment]
I've added support for all configuration parameters + config file overriding.

Examples:

`launcher.exe --config=config.json --parameters=ema-fast:50,ema-slow:30 --environment=backtesting --parameters=some-forgotten-parameter:100`

in this case, `parameters` arguments are merged together so we got single `parameters` object:
```json
{
"ema-fast": 50,
"ema-slow":30,
"some-forgotten-parameter":100
}
```

Parameters are automatically parsed:
`launcher.exe --live-data-port=8020`

This will automatically parse "8020" as number.

File: Configuration/Config.cs [hunk 1]
```diff
@@ -30,7 +30,45 @@ namespace QuantConnect.Configuration
...
+            }
+
+            var jsonArguments = JsonConvert.DeserializeObject(JsonConvert.SerializeObject(cliArguments));
+            
+            Settings.Value.Merge(jsonArguments, new JsonMergeSettings
+            {
+                MergeArrayHandling = MergeArrayHandling.Union
+            });
+            
+            Log.Trace(Settings.Value.ToString());
```
Comment (StefanoRaggi) [inline Configuration/Config.cs]: We should probably remove this line to prevent user credentials ending up in the log file

**Comment by mamatopian** [comment]
@jaredbroad @StefanoRaggi any other things to fix guys?

**Comment by jaredbroad** [comment]
Nice work @mamatopian

**Pull request merged**